packages/web/src/js/__tests__/FileUploader.test.ts

```diff
@@ -96,4 +97,80 @@ describe('FileUploader', () => {
       expect(fileUploaderInstance.fileQueueLimit).toBe(10); // Default file queue limit
     });
   });
+
+  describe('Supported and Unsupported files', () => {
+    let fileUploader: FileUploader;
+
+    beforeEach(() => {
+      fixtureEl.innerHTML = `
+      <div class="file-uploader" data-max-file-size="10000000" data-file-queue-limit="10">
+        <div class="file-uploader__wrapper" data-spirit-element="wrapper">
+          <ul class="file-uploader__list" data-spirit-element="list"></ul>
+          <input type="file" name="attachment" multiple data-spirit-element="input" accept="image/*" />
+          <div class="file-uploader__drop-zone" data-spirit-element="dropZone">Drag and drop files here</div>
+        </div>
+      </div>
+    `;
+
+      fileUploader = new FileUploader(SelectorEngine.findOne('.file-uploader')!);
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    describe('checkAllowedFileType()', () => {
+      it('should allow a supported file type', () => {
+        const file = new File(['content'], 'example.jpg', { type: 'image/jpeg' });
+        expect(() => fileUploader.checkAllowedFileType(file)).not.toThrow();
+      });
+
+      it('should throw an error for an unsupported file type', () => {
+        const file = new File(['content'], 'example.txt', { type: 'text/plain' });
+        expect(() => fileUploader.checkAllowedFileType(file)).toThrowError(
+          `The file "example.txt" is not supported. Please ensure you are uploading a supported file format.`,
+        );
+      });
+    });
+  });
+
+  describe('FileUploader isDismissible', () => {
+    let fileUploader: FileUploader;
+
+    beforeEach(() => {
+      fixtureEl.innerHTML = `
+      <div class="file-uploader" data-file-queue-limit="2">
+        <div class="file-uploader__wrapper" data-spirit-element="wrapper">
+          <ul class="file-uploader__list" data-spirit-element="list"></ul>
+          <input type="file" name="attachment" multiple data-spirit-element="input" />
+          <div class="file-uploader__drop-zone" data-spirit-element="dropZone">Drag and drop files here</div>
+        </div>
+      </div>
+    `;
+
+      fileUploader = new FileUploader(SelectorEngine.findOne('.file-uploader')!);
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should hides the drop zone when the file queue limit is reached', async () => {
+      const dropZone = fixtureEl.querySelector('[data-spirit-element="dropZone"]') as HTMLElement;
+      const file1 = new File([''], 'test1.txt', { type: 'text/plain', lastModified: Date.now() });
```

Comment:
Can we please end with using actual date in tests, when it is not necessary? I know that we are already using this somewhere but actually, this is a bad habit to continue with this.

As I mentioned earlier, this can cause some errors in the future when we start working with `lastModified`. Another point is that the test is slightly more performant when they do not need to recalculate actual DateTime.

Reply:
I created issue for that https://jira.lmc.cz/browse/DS-769. I spend too much time for creating unit tests. So that we don't stop ourselves anymore

Reply:
done